### ISSUE TITLE
TestReadHeader/equals_in_name fails on Go1.20

### ISSUE THREAD

**Issue by zhsj** [jhillyerd/enmime#269]
What I did:

```
$ go version
go version go1.20rc3 linux/amd64
```

```
go test -run TestReadHeader/equals_in_name
```

What I expected: PASS

What I got:

```
--- FAIL: TestReadHeader (0.00s)
    --- FAIL: TestReadHeader/equals_in_name (0.00s)
        header_test.go:325: malformed MIME header line: name=value:text
FAIL
```

Release or branch I am using: master

**Comment by jhillyerd** [comment]
If you need a quick workaround until this is fixed, you can delete this `equals in name` test case: 

https://github.com/jhillyerd/enmime/blob/04d6a5bdb98ce6d86fb7ba3c61ff61d0a65b835b/header_test.go#L247

Well formed mail should not have `=` in MIME header names, but you may see the rare parsing failure from it.

**Comment by jhillyerd** [comment]
I've got a fix proposed in #270 -- it's going to make enmime quite a bit more picky about what characters are allowed in headers.  When it encounters it will simply discard the header (and add an Error to the Part), rather than fail outright.  This is probably the best default behavior, but I can imagine some users may want some sort of override or perhaps a "bad headers" map they can inspect.
